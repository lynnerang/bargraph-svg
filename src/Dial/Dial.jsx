import React, { Component } from "react";
import { Knob } from "react-rotary-knob";

class Dial extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
      startVal: 0,
      stopVal: 0
    };
  }

  render() {
    console.log(this.state.startVal)
    console.log("val: " + this.props.value)

    let { value, onDialChange } = this.props;

    return (
      <Knob
        min={0}
        max={1000}
        onStart={() => this.setState({ startVal: value })}
        onStop={() => onDialChange(this.state.startVal - value)}
        style={{ display: "inline-block", width: "200", height: "200" }}
        width={200}
        height={200}
        unlockDistance={0}
        preciseMode={false}
      />
      
    );
  }
}

export default Dial;