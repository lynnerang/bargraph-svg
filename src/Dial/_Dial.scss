.dial-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 12px;
}

.dial-bg {
  height: 120px;
  width: 120px;
  background-image: url('../images/dial.png');
  background-size: cover;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  h2 {
    font-size: 18px;
  }

  p {
    font-size: 12px;
  }
}

button {
  width: 50px;
  height: 50px;
  background: none;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.fas {
  font-size: 80px;
  line-height: 50px;
  color: #3f51b5;

  &:hover {
    color: rgb(155, 155, 235);
    cursor: pointer;
  }
}

.fa-caret-down {
  margin-top: -5px;
}


