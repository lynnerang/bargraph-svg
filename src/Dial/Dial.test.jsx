import React from 'react';
import ReactDOM from 'react-dom';
import Dial from './Dial';
import { mount } from 'enzyme';

describe('Dial', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Dial />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot with name of "Coarse"', () => {
    const wrapper = mount(<Dial name="Coarse"/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with name of "Fine"', () => {
    const wrapper = mount(<Dial name="Fine" />);
    expect(wrapper).toMatchSnapshot();
  })

  it('calls onDialChange with a value of 100 when name is "Coarse" and the up button is clicked', () => {
    const mockOnDialChange = jest.fn();
    const wrapper = mount(<Dial name="Coarse" onDialChange={mockOnDialChange} />);
    
    wrapper.find('.fa-caret-up').simulate('click');

    expect(mockOnDialChange).toHaveBeenCalledWith(100);
  })

  it('calls onDialChange with a value of -100 when name is "Coarse" and the down button is clicked', () => {
    const mockOnDialChange = jest.fn();
    const wrapper = mount(<Dial name="Coarse" onDialChange={mockOnDialChange} />);

    wrapper.find('.fa-caret-down').simulate('click');

    expect(mockOnDialChange).toHaveBeenCalledWith(-100);
  })

  it('calls onDialChange with a value of 1 when name is "Fine" and the up button is clicked', () => {
    const mockOnDialChange = jest.fn();
    const wrapper = mount(<Dial name="Fine" onDialChange={mockOnDialChange} />);

    wrapper.find('.fa-caret-up').simulate('click');

    expect(mockOnDialChange).toHaveBeenCalledWith(1);
  })

  it('calls onDialChange with a value of -1 when name is "Fine" and the down button is clicked', () => {
    const mockOnDialChange = jest.fn();
    const wrapper = mount(<Dial name="Fine" onDialChange={mockOnDialChange} />);

    wrapper.find('.fa-caret-down').simulate('click');

    expect(mockOnDialChange).toHaveBeenCalledWith(-1);
  })

});