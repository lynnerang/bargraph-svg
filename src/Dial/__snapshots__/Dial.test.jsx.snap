// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dial matches the snapshot with name of "Coarse" 1`] = `
<Dial
  name="Coarse"
>
  <article
    className="dial-wrapper"
  >
    <button>
      <i
        className="fas fa-caret-up"
        onClick={[Function]}
      />
    </button>
    <div
      className="dial-bg"
    >
      <h2
        className="dial-name"
      >
        Coarse
      </h2>
      <p
        className="dial-label"
      >
        (
        100
         mV)
      </p>
    </div>
    <button>
      <i
        className="fas fa-caret-down"
        onClick={[Function]}
      />
    </button>
  </article>
</Dial>
`;

exports[`Dial matches the snapshot with name of "Fine" 1`] = `
<Dial
  name="Fine"
>
  <article
    className="dial-wrapper"
  >
    <button>
      <i
        className="fas fa-caret-up"
        onClick={[Function]}
      />
    </button>
    <div
      className="dial-bg"
    >
      <h2
        className="dial-name"
      >
        Fine
      </h2>
      <p
        className="dial-label"
      >
        (
        1
         mV)
      </p>
    </div>
    <button>
      <i
        className="fas fa-caret-down"
        onClick={[Function]}
      />
    </button>
  </article>
</Dial>
`;
